playground/src/server/index.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import assert from 'assert';
+import fs from 'fs';
+import path from 'path';
+import crypto from 'crypto';
+
+import Fastify from 'fastify';
+
+import {
+    fetcher,
+} from './fetcher';
+import {
+    fromJwk as createSignerFromJwk,
+} from './signer';
+import {
+    WasmResponse,
+    createWorker,
+} from './wasmFunctions';
+
+const wasmBuffer = fs.readFileSync(path.resolve(__dirname, '..', '..', 'cloudflare_worker', 'pkg', 'cloudflare_worker_bg.wasm'));
+
+async function headerIntegrityGet(_url: string): Promise<WasmResponse> {
+    return {
+        body: [],
+        headers: [],
+        status: 404,
+    };
+}
+
+async function headerIntegrityPut() {
+}
+
+function createSrp(outerUrl: string, innerUrl: string) {
+    return `
+        <link rel=prefetch href="${outerUrl}">
+        <p>${innerUrl}<p>
+        <a href="${outerUrl}">SXG</a>
+        <a href="${innerUrl}">Non-SXG</a>
+    `
+}
+
+export const SXG_CONFIG = `
+cert_url_dirname: ".well-known/sxg-certs"
+forward_request_headers:
+  - user-agent
+  - cf-ipcountry
+reserved_path: ".sxg"
+respond_debug_info: false
+strip_request_headers: []
+strip_response_headers:
+  - set-cookie
+  - strict-transport-security
+validity_url_dirname: ".well-known/sxg-validity"
+`;
+export async function startSxgServer({
+    certificatePem,
+    privateKeyJwk,
+    privateKeyPem,
+}: {
+    certificatePem: string,
+    privateKeyJwk: Object,
+    privateKeyPem: string,
+}) {
+    const signer = createSignerFromJwk((crypto.webcrypto as any).subtle, privateKeyJwk);
+    const worker = await createWorker(wasmBuffer, SXG_CONFIG, certificatePem, certificatePem);
+    const sxgList: WasmResponse[] = [];
+    async function createSxgIntoList(innerUrl: string, certOrigin: string) {
+        const sxgPayload = await(fetcher({
+            url: innerUrl,
+            body: [],
+            method: 'Get',
+            headers: [],
+        }));
+        const sxg = await worker.createSignedExchange(
```

Comment:
Is it possible to call `handleRequest` from cloudflare_worker/worker/src/index.ts instead? Or alternatively replicate the `processHTML` calls that it does. OK to defer this to a future PR.

Reply:
Will do this in a future PR.